src/v0/destinations/twitter_ads/transform.js

```diff
@@ -0,0 +1,227 @@
+const sha256 = require('sha256');
+const crypto = require('crypto');
+const oauth1a = require('oauth-1.0a');
+const {
+  constructPayload,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  isDefinedAndNotNull,
+  simpleProcessRouterDest,
+} = require('../../util');
+const { EventType } = require('../../../constants');
+const {
+  ConfigCategories,
+  mappingConfig,
+  BASE_URL,
+  authHeaderConstant
+} = require('./config');
+
+const { InstrumentationError, OAuthSecretError, ConfigurationError } = require('../../util/errorTypes');
+const { JSON_MIME_TYPE } = require('../../util/constant');
+
+const getOAuthFields = ({ secret }) => {
+  if (!secret) {
+    throw new OAuthSecretError('[TWITTER ADS]:: OAuth - access keys not found');
+  }
+  const oAuthObject = {
+    consumerKey: secret.consumerKey,
+    consumerSecret: secret.consumerSecret,
+    accessToken: secret.accessToken,
+    accessTokenSecret: secret.accessTokenSecret
+  };
+  return oAuthObject;
+};
+
+function getAuthHeaderForRequest(request, oAuthObject) {
+    const oauth = oauth1a({
+      consumer: { key: oAuthObject.consumerKey, secret: oAuthObject.consumerSecret },
+      signature_method: 'HMAC-SHA1',
+      hash_function(base_string, k) {
+        return crypto
+            .createHmac('sha1', k)
+            .update(base_string)
+            .digest('base64')
+      },
+    })
+
+    const authorization = oauth.authorize(request, {
+      key: oAuthObject.accessToken,
+      secret: oAuthObject.accessTokenSecret,
+    });
+
+    return oauth.toHeader(authorization);
+}
+
+// build final response
+function buildResponse(message, requestJson, metadata, endpointUrl) {
+  const response = defaultRequestConfig();
+  response.endpoint = endpointUrl;
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.body.JSON.conversions = [removeUndefinedAndNullValues(requestJson)];
+  // required to be in accordance with oauth package
+  const request = {
+    url: response.endpoint,
+    method: response.method,
+    body: response.body.JSON
+  };
+  const oAuthObject = getOAuthFields(metadata);
+  let authHeader = getAuthHeaderForRequest(request, oAuthObject).Authorization;
+  if(message.properties.testModeEnable === true) {
```

Comment:
I don't remember we discussing about `testModeEnable` and this certainly cannot be taken from the payload rather this kind of thing should a configuration setting.